Description: Avoid video drivers that probe PCI space
 video_bochs and video_cirrus break some devices by probing PCI space.
 Disable them as a short-term fix for Ubuntu 10.10.
Author: Colin Watson <cjwatson@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/641259
Forwarded: no
Last-Update: 2010-10-06

Index: b/util/grub.d/00_header.in
===================================================================
--- a/util/grub.d/00_header.in
+++ b/util/grub.d/00_header.in
@@ -91,6 +91,20 @@
 else
     # Insert all available backends; GRUB will use the most appropriate.
     for backend in $(cat "${GRUB_PREFIX}/video.lst"); do
+	# video_bochs and video_cirrus require probing PCI space, and some
+	# machines don't seem to like this.  These are generally
+	# non-essential at least for i386-pc, so disable them as a
+	# short-term fix for Ubuntu 10.10.
+	case "${backend}" in
+	    video_bochs|video_cirrus)
+		cpu="$(uname -m)"
+		case "${cpu}" in
+		    i[3456]86|x86_64)
+			[ -d /sys/firmware/efi ] || continue
+		    ;;
+		esac
+	    ;;
+	esac
 	cat <<EOF
   insmod ${backend}
 EOF
